crates/flipperzero/src/notification/mod.rs

```diff
@@ -0,0 +1,308 @@
+//! Furi notifications.
+
+use core::ffi::c_char;
+
+use flipperzero_sys as sys;
+use flipperzero_sys::furi::UnsafeRecord;
+
+///Default notification messages.
+pub mod messages;
+///Default notification notes.
+pub mod notes;
+///Default notification sequences.
+pub mod sequences;
+
+const RECORD_NOTIFICATION: *const c_char = sys::c_string!("notification");
+
+/// A handle to the Notification app.
+pub struct NotificationApp {
+    data: UnsafeRecord<sys::NotificationApp>,
+}
+
+impl NotificationApp {
+    /// Obtains a handle to the Notifications app.
+    pub fn open() -> Self {
+        Self {
+            data: unsafe { UnsafeRecord::open(RECORD_NOTIFICATION) },
+        }
+    }
+
+    /// Runs a notification sequence.
+    ///
+    /// #Safety
+    /// Due to how rust interacts with the firmware this function is not safe to use at any time
+    /// where the application might exit directly afterwards as the rust runtime will free the
+    /// sequence before the firmware has finished reading it. At any time where this is an issue
+    /// `notify_blocking` should be used instead..
+    pub fn notify(&mut self, sequence: &'static NotificationSequence) {
+        unsafe { sys::notification_message(self.data.as_ptr(), sequence.to_sys()) };
+    }
+
+    /// Runs a notification sequence and blocks the thread.
+    pub fn notify_blocking(&mut self, sequence: &'static NotificationSequence) {
+        unsafe { sys::notification_message_block(self.data.as_ptr(), sequence.to_sys()) };
+    }
+}
+
+#[repr(u8)]
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum Light {
+    Off = 0,
+
+    Red = 0b0001,
+    Green = 0b0010,
+    Blue = 0b0100,
+    Backlight = 0b1000,
+
+    Cyan = 0b0110,
+    Magenta = 0b0101,
+    Yellow = 0b0011,
+
+    White = 0b0111,
+}
```

Comment:
We might want to consider using the [`bitflags`](https://crates.io/crates/bitflags) crate, rather than rolling our own bitflags-like types.

Reply:
Thanks for the tip! I had to use v1.0 though as v2.0 does not expose the generated constants in the docs for some reason.